--- xen-4.1.1/tools/libxl/libxl_dm.c.orig	2013-02-21 05:11:16.392000000 +0100
+++ xen-4.1.1/tools/libxl/libxl_dm.c	2013-02-21 05:11:20.860000000 +0100
@@ -458,7 +458,7 @@
     libxl_domain_create_info c_info;
     libxl_domain_build_info b_info;
     libxl_domain_build_state state;
-    uint32_t domid;
+    uint32_t domid = 0;
     char **args;
     struct xs_permissions perm[2];
     xs_transaction_t t;
