--- xen-4.0.1/tools/blktap/lib/blktaplib.h.orig	2010-08-25 11:22:07.000000000 +0100
+++ xen-4.0.1/tools/blktap/lib/blktaplib.h	2010-08-29 20:57:11.000000000 +0100
@@ -195,8 +195,10 @@
 	pid_t     pid;
 } msg_pid_t;
 
+#ifndef READ
 #define READ 0
 #define WRITE 1
+#endif

 /*Control Messages between manager and tapdev*/
 #define CTLMSG_PARAMS      1
--- xen-4.0.1/tools/blktap2/include/blktaplib.h.orig	2010-08-25 11:22:07.000000000 +0100
+++ xen-4.0.1/tools/blktap2/include/blktaplib.h	2010-08-29 21:50:50.000000000 +0100
@@ -197,8 +197,10 @@
 	int       uuid_len;
 } msg_lock_t;
 
+#ifndef READ
 #define READ 0
 #define WRITE 1
+#endif
 
 /*Control Messages between manager and tapdev*/
 #define CTLMSG_PARAMS          1
