--- xen-4.13.0/tools/ocaml/libs/xc/xenctrl_stubs.c.orig	2019-12-17 14:23:09.000000000 +0000
+++ xen-4.13.0/tools/ocaml/libs/xc/xenctrl_stubs.c	2020-01-21 19:24:49.508024245 +0000
@@ -46,7 +46,7 @@
 #define Val_none (Val_int(0))
 
 #define string_of_option_array(array, index) \
-	((Field(array, index) == Val_none) ? NULL : String_val(Field(Field(array, index), 0)))
+	((Field(array, index) == Val_none) ? NULL : (char *) Bp_val(Field(Field(array, index), 0)))
 
 /* maybe here we should check the range of the input instead of blindly
  * casting it to uint32 */
@@ -505,7 +505,7 @@
 	ret = xc_vcpu_getcontext(_H(xch), _D(domid), Int_val(cpu), &ctxt);
 
 	context = caml_alloc_string(sizeof(ctxt));
-	memcpy(String_val(context), (char *) &ctxt.c, sizeof(ctxt.c));
+	memcpy((char *) Bp_val(context), (char *) &ctxt.c, sizeof(ctxt.c));
 
 	CAMLreturn(context);
 }
@@ -684,7 +684,7 @@
 		conring_size = size;
 
 	ring = caml_alloc_string(count);
-	memcpy(String_val(ring), str, count);
+	memcpy((char *) Bp_val(ring), str, count);
 	free(str);
 
 	CAMLreturn(ring);
@@ -695,7 +695,7 @@
 	CAMLparam2(xch, keys);
 	int r;
 
-	r = xc_send_debug_keys(_H(xch), String_val(keys));
+	r = xc_send_debug_keys(_H(xch), (char *) Bp_val(keys));
 	if (r)
 		failwith_xc(_H(xch));
 	CAMLreturn(Val_unit);
@@ -855,7 +855,7 @@
 	}
 
 	for (r = 0; r < 4; r++)
-		out_config[r] = (c_config[r]) ? String_val(Field(Field(array, r), 0)) : NULL;
+		out_config[r] = (c_config[r]) ? (char *) Bp_val(Field(Field(array, r), 0)) : NULL;
 
 	r = xc_cpuid_set(_H(xch), _D(domid),
 			 c_input, (const char **)c_config, out_config);
--- xen-4.13.0/tools/ocaml/libs/xb/xs_ring_stubs.c.orig	2019-12-17 14:23:09.000000000 +0000
+++ xen-4.13.0/tools/ocaml/libs/xb/xs_ring_stubs.c	2020-01-21 23:51:35.473330934 +0000
@@ -44,7 +44,7 @@
 	CAMLlocal1(ml_result);
 
 	struct mmap_interface *interface = GET_C_STRUCT(ml_interface);
-	char *buffer = String_val(ml_buffer);
+	char *buffer = (char *) Bp_val(ml_buffer);
 	int len = Int_val(ml_len);
 	int result;
 
@@ -103,7 +103,7 @@
 	CAMLlocal1(ml_result);
 
 	struct mmap_interface *interface = GET_C_STRUCT(ml_interface);
-	char *buffer = String_val(ml_buffer);
+	char *buffer = (char *) Bp_val(ml_buffer);
 	int len = Int_val(ml_len);
 	int result;
 
--- xen-4.13.0/tools/ocaml/libs/xb/xenbus_stubs.c.orig	2019-12-17 14:23:09.000000000 +0000
+++ xen-4.13.0/tools/ocaml/libs/xb/xenbus_stubs.c	2020-01-22 00:04:09.443168991 +0000
@@ -65,7 +65,7 @@
 	};
 
 	ret = caml_alloc_string(sizeof(struct xsd_sockmsg));
-	memcpy(String_val(ret), &xsd, sizeof(struct xsd_sockmsg));
+	memcpy((char *) Bp_val(ret), &xsd, sizeof(struct xsd_sockmsg));
 
 	CAMLreturn(ret);
 }
