--- xen-4.11.0/tools/xenmon/Makefile.orig	2018-07-09 14:47:19.000000000 +0100
+++ xen-4.11.0/tools/xenmon/Makefile	2018-09-10 21:13:15.200655105 +0100
@@ -32,7 +32,7 @@
 	$(INSTALL_DIR) $(DESTDIR)$(sbindir)
 	$(INSTALL_PROG) xenbaked $(DESTDIR)$(sbindir)/xenbaked
 	$(INSTALL_PROG) xentrace_setmask  $(DESTDIR)$(sbindir)/xentrace_setmask
-	$(INSTALL_PROG) xenmon.py  $(DESTDIR)$(sbindir)/xenmon
+	$(INSTALL_PYTHON_PROG) xenmon.py  $(DESTDIR)$(sbindir)/xenmon
 
 .PHONY: uninstall
 uninstall:
--- xen-4.11.0/tools/python/Makefile.orig	2018-07-09 14:47:19.000000000 +0100
+++ xen-4.11.0/tools/python/Makefile	2018-09-10 21:21:07.097979007 +0100
@@ -20,8 +20,8 @@
 		setup.py install --record $(INSTALL_LOG) $(PYTHON_PREFIX_ARG) \
 		--root="$(DESTDIR)" --force
 
-	$(INSTALL_PROG) scripts/convert-legacy-stream $(DESTDIR)$(LIBEXEC_BIN)
-	$(INSTALL_PROG) scripts/verify-stream-v2 $(DESTDIR)$(LIBEXEC_BIN)
+	$(INSTALL_PYTHON_PROG) scripts/convert-legacy-stream $(DESTDIR)$(LIBEXEC_BIN)
+	$(INSTALL_PYTHON_PROG) scripts/verify-stream-v2 $(DESTDIR)$(LIBEXEC_BIN)
 
 .PHONY: uninstall
 uninstall:
--- xen-4.11.0/tools/python/install-wrap.orig	2018-07-09 14:47:19.000000000 +0100
+++ xen-4.11.0/tools/python/install-wrap	2018-09-11 20:09:57.803655357 +0100
@@ -44,7 +44,7 @@
 destf="$dest"
 for srcf in ${srcs}; do
 	if test -d "$dest"; then
-		destf="$dest/${srcf%%*/}"
+		destf="$dest/${srcf##*/}"
 	fi
 	org="$(sed -n '2q; /^#! *\/usr\/bin\/env python *$/p' $srcf)"
 	if test "x$org" = x; then
--- xen-4.11.0/tools/misc/xencov_split.orig	2018-07-09 14:47:19.000000000 +0100
+++ xen-4.11.0/tools/misc/xencov_split	2018-09-18 21:56:07.397893895 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 import sys, os, os.path as path, struct, errno
 from optparse import OptionParser
