--- xen-4.2.4/tools/qemu-xen-traditional/texi2pod.pl.orig	2014-06-16 04:24:47.658954702 +0200
+++ xen-4.2.4/tools/qemu-xen-traditional/texi2pod.pl	2014-06-16 04:25:04.382295454 +0200
@@ -340,6 +340,8 @@
 $sects{NAME} = "$fn \- $tl\n";
 $sects{FOOTNOTES} .= "=back\n" if exists $sects{FOOTNOTES};
 
+print "=encoding utf8\n";
+
 for $sect (qw(NAME SYNOPSIS DESCRIPTION OPTIONS ENVIRONMENT FILES
 	      BUGS NOTES FOOTNOTES SEEALSO AUTHOR COPYRIGHT)) {
     if(exists $sects{$sect}) {
