diff -ru xen-4.1.0.orig/Makefile xen-4.1.0/Makefile
--- xen-4.4.0/m4/fetcher.m4.orig	2014-06-24 01:08:52.593933406 +0200
+++ xen-4.4.0/m4/fetcher.m4	2014-06-24 01:09:29.337842284 +0200
@@ -1,14 +1,4 @@
 AC_DEFUN([AX_CHECK_FETCHER], [
-AC_PATH_PROG([WGET],[wget], [no])
-AS_IF([test x"$WGET" != x"no"], [
-    FETCHER="$WGET -c -O"
-], [
-    AC_PATH_PROG([FTP],[ftp], [no])
-    AS_IF([test x"$FTP" != x"no"], [
-        FETCHER="$FTP -o"
-    ], [
-        AC_MSG_ERROR([cannot find wget or ftp])
-    ])
-])
+FETCHER="false"
 AC_SUBST(FETCHER)
 ])
--- a/config/StdGNU.mk.orig
+++ b/config/StdGNU.mk
@@ -19,7 +19,7 @@ OBJDUMP    = $(CROSS_COMPILE)objdump
 SIZEUTIL   = $(CROSS_COMPILE)size
 
 # Allow git to be wrappered in the environment
-GIT        ?= git
+GIT        ?= false
 
 INSTALL      = install
 INSTALL_DIR  = $(INSTALL) -d -m0755 -p
