--- a/tools/libxl/libxl_pci.c
+++ b/tools/libxl/libxl_pci.c
@@ -1421,6 +1421,17 @@ static int do_pci_add(libxl__gc *gc, uin
             rc = ERROR_FAIL;
             goto out;
         }
+    } else if (libxl_is_stubdom(ctx, domid, NULL)) {
+        /* Allow acces to MSI enable flag in PCI config space for the stubdom */
+        if ( sysfs_write_bdf(gc, SYSFS_PCIBACK_DRIVER"/allow_interrupt_control",
+                             pcidev) < 0 ) {
+            if ( sysfs_write_bdf(gc, SYSFS_PCIBACK_DRIVER"/allow_msi_enable",
+                                 pcidev) < 0 ) {
+                LOGD(ERROR, domainid, "Setting allow_msi_enable for device");
+                rc = ERROR_FAIL;
+                goto out;
+            }
+        }
     }
 
 out_no_irq:
