--- a/tools/libxl/libxl_create.c
+++ b/tools/libxl/libxl_create.c
@@ -365,6 +365,18 @@ int libxl__domain_build_info_setdefault(
             libxl_defbool_setdefault(&b_info->u.hvm.sdl.opengl, false);
         }
 
+        libxl_defbool_setdefault(&b_info->u.hvm.qubes_gui.enable, false);
+        if (libxl_defbool_val(b_info->u.hvm.qubes_gui.enable)) {
+            if (b_info->u.hvm.qubes_gui.domname) {
+                if (libxl__resolve_domid(gc,
+                                         b_info->u.hvm.qubes_gui.domname,
+                                         &b_info->u.hvm.qubes_gui.domid) < 0) {
+                    LOG(ERROR, "Qubes GUI domain not found.");
+                    return ERROR_INVAL;
+                }
+            }
+        }
+
         if (libxl_defbool_val(b_info->u.hvm.spice.enable)) {
             libxl_defbool_setdefault(&b_info->u.hvm.spice.disable_ticketing,
                                      false);
--- a/tools/libxl/libxl_dm.c
+++ b/tools/libxl/libxl_dm.c
@@ -1272,6 +1272,13 @@ static int libxl__build_device_model_arg
             flexarray_append(dm_args, "-net");
             flexarray_append(dm_args, "none");
         }
+
+        if (libxl_defbool_val(b_info->u.hvm.qubes_gui.enable)) {
+            flexarray_append_pair(dm_args, "-display",
+                                  GCSPRINTF("qubes-gui,domid=%u,log_level=%i",
+                                            b_info->u.hvm.qubes_gui.domid,
+                                            b_info->u.hvm.qubes_gui.log_level));
+        }
     } else {
         if (!sdl && !vnc) {
             flexarray_append(dm_args, "-nographic");
--- a/tools/libxl/libxl_types.idl
+++ b/tools/libxl/libxl_types.idl
@@ -301,6 +301,13 @@ libxl_sdl_info = Struct("sdl_info", [
     ("xauthority",    string),
     ])
 
+libxl_qubes_gui_info = Struct("qubes_gui_info", [
+    ("enable", libxl_defbool),
+    ("domid", libxl_domid),
+    ("domname", string),
+    ("log_level", integer),
+    ])
+
 libxl_dominfo = Struct("dominfo",[
     ("uuid",        libxl_uuid),
     ("domid",       libxl_domid),
@@ -557,6 +564,8 @@ libxl_domain_build_info = Struct("domain
                                        ("serial_list",      libxl_string_list),
                                        ("rdm", libxl_rdm_reserve),
                                        ("rdm_mem_boundary_memkb", MemKB),
+
+                                       ("qubes_gui", libxl_qubes_gui_info),
                                        ])),
                  ("pv", Struct(None, [("kernel", string),
                                       ("slack_memkb", MemKB),
