TODO: upstream?
--- a/tools/libxl/libxl_mem.c
+++ b/tools/libxl/libxl_mem.c
@@ -4493,7 +4493,7 @@ int libxl_domain_need_memory(libxl_ctx *
         if (libxl_defbool_val(b_info->device_model_stubdomain)) {
             switch (b_info->stubdomain_version) {
             case LIBXL_STUBDOMAIN_VERSION_MINIOS:
-                *need_memkb += 32 * 1024;
+                *need_memkb += 28 * 1024;
                 break;
             case LIBXL_STUBDOMAIN_VERSION_LINUX:
                 *need_memkb += LIBXL_LINUX_STUBDOM_MEM * 1024;
@@ -4501,6 +4501,7 @@ int libxl_domain_need_memory(libxl_ctx *
             default:
                 abort();
             }
+            *need_memkb += b_info->video_memkb;
         }
         break;
     case LIBXL_DOMAIN_TYPE_PV:
