--- xen-4.7.0/tools/libxl/libxl_create.c.orig	2016-07-15 20:01:43.126000000 +0300
+++ xen-4.7.0/tools/libxl/libxl_create.c	2016-07-15 20:03:23.445000000 +0300
@@ -652,6 +652,9 @@
                     GCSPRINTF("%s/data", dom_path),
                     rwperm, ARRAY_SIZE(rwperm));
     libxl__xs_mknod(gc, t,
+                    GCSPRINTF("%s/error", dom_path),
+                    rwperm, ARRAY_SIZE(rwperm));
+    libxl__xs_mknod(gc, t,
                     GCSPRINTF("%s/drivers", dom_path),
                     rwperm, ARRAY_SIZE(rwperm));
     libxl__xs_mknod(gc, t,
