diff --git a/tools/libxl/libxl_dm.c b/tools/libxl/libxl_dm.c
index f2dc5696b9..f044f2566c 100644
--- a/tools/libxl/libxl_dm.c
+++ b/tools/libxl/libxl_dm.c
@@ -2274,7 +2274,7 @@ void libxl__spawn_stub_dm(libxl__egc *eg
     dm_config->b_info.target_memkb = dm_config->b_info.max_memkb;

     dm_config->b_info.max_grant_frames = guest_config->b_info.max_grant_frames;
-    dm_config->b_info.max_maptrack_frames = 0;
+    dm_config->b_info.max_maptrack_frames = guest_config->b_info.max_maptrack_frames;

     dm_config->b_info.u.pv.features = "";

