From: Ian Jackson <ian.jackson@citrix.com>
Date: Thu, 20 Sep 2018 18:10:14 +0100
Subject: Do not build the instruction emulator

Signed-off-by: Ian Jackson <ian.jackson@citrix.com>
---
 tools/fuzz/Makefile | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/tools/fuzz/Makefile b/tools/fuzz/Makefile
index 85213dc..6b7970b 100644
--- a/tools/fuzz/Makefile
+++ b/tools/fuzz/Makefile
@@ -3,7 +3,10 @@ include $(XEN_ROOT)/tools/Rules.mk
 
 SUBDIRS-y :=
 SUBDIRS-y += libelf
-SUBDIRS-y += x86_instruction_emulator
+
+#SUBDIRS-y += x86_instruction_emulator
+# This does not compile with Debian's userland compile options,
+# which include fortify etc.
 
 .PHONY: all clean distclean install uninstall
 all clean distclean install uninstall: %: subdirs-%
