diff --git a/xen/Makefile b/xen/Makefile
index 46c8177..9fa1dda 100644
--- a/xen/Makefile
+++ b/xen/Makefile
@@ -207,7 +207,7 @@ CFLAGS += -fomit-frame-pointer
 endif
 
 CFLAGS += -nostdinc -fno-builtin -fno-common
-CFLAGS += -Werror -Wredundant-decls -Wno-pointer-arith
+CFLAGS += -Werror -Wredundant-decls -Wno-pointer-arith -Wno-error=array-bounds -Wno-error=address
 $(call cc-option-add,CFLAGS,CC,-Wvla)
 CFLAGS += -pipe -D__XEN__ -include $(BASEDIR)/include/xen/config.h
 CFLAGS-$(CONFIG_DEBUG_INFO) += -g
