Index: xen-hvm/tools/qemu-xen-traditional/xenstore.c
===================================================================
--- xen-hvm.orig/tools/qemu-xen-traditional/xenstore.c
+++ xen-hvm/tools/qemu-xen-traditional/xenstore.c
@@ -248,7 +248,10 @@ static void xenstore_get_backend_path(ch
     char *frontend_backend_path=0;
     char *backend_frontend_path=0;
     char *frontend_doublecheck=0;
+    char *domid_backend_str=0;
+    char *domid_backend_path=0;
     int len;
+    int domid_backend = 0;
     const char *frontend_idpath_slash;
 
     /* clear out return value for if we error out */
@@ -269,7 +272,13 @@ static void xenstore_get_backend_path(ch
                   frontend_path)
         == -1) goto out;
 
+    if (pasprintf(&domid_backend_path, "%s/backend-id",
+                  frontend_path)
+        == -1) goto out;
+
     bpath = xs_read(xsh, XBT_NULL, frontend_backend_path, &len);
+    domid_backend_str = xs_read(xsh, XBT_NULL, domid_backend_path, &len);
+    domid_backend = atoi(domid_backend_str);
 
     /* now we must check that the backend is intended for use
      * by this frontend, since the frontend's /backend xenstore node
