diff -r 4683409fac3d xen/common/page_alloc.c
--- a/xen/common/page_alloc.c	Sun Sep 04 23:49:56 2011 +0200
+++ b/xen/common/page_alloc.c	Thu Sep 08 00:51:24 2011 +0200
@@ -336,7 +336,7 @@ static struct page_info *alloc_heap_page
      * Others try tmem pools then fail.  This is a workaround until all
      * post-dom0-creation-multi-page allocations can be eliminated.
      */
-    if ( opt_tmem && ((order == 0) || (order >= 9)) &&
+    if ( ((order == 0) || (order >= 9)) &&
          (total_avail_pages <= midsize_alloc_zone_pages) &&
          tmem_freeable_pages() )
         goto try_tmem;
@@ -567,7 +567,7 @@ static void free_heap_pages(
     avail[node][zone] += 1 << order;
     total_avail_pages += 1 << order;
 
-    if ( opt_tmem )
+    if ( 1 )
         midsize_alloc_zone_pages = max(
             midsize_alloc_zone_pages, total_avail_pages / MIDSIZE_ALLOC_FRAC);
 
