diff -rup xen-3.3.0-orig/tools/xenstore/xenstore_client.c xen-3.3.0-new/tools/xenstore/xenstore_client.c
--- xen-3.3.0-orig/tools/xenstore/xenstore_client.c	2008-08-22 10:49:09.000000000 +0100
+++ xen-3.3.0-new/tools/xenstore/xenstore_client.c	2008-08-29 12:20:48.000000000 +0100
@@ -589,7 +589,9 @@ main(int argc, char **argv)
 
     xsh = socket ? xs_daemon_open() : xs_domain_open();
     if (xsh == NULL)
-	err(1, socket ? "xs_daemon_open" : "xs_domain_open");
+	xsh = socket ? xs_domain_open() : xs_daemon_open();
+    if (xsh == NULL)
+	err(1, "xs_*_open");
 
 again:
     if (transaction) {
