--- xen-3.4.3/tools/python/xen/xend/XendDomainInfo.py.orig	2010-09-15 17:24:23.275074023 +0200
+++ xen-3.4.3/tools/python/xen/xend/XendDomainInfo.py	2010-09-15 18:15:03.779077596 +0200
@@ -2525,6 +2525,8 @@
             # the various headrooms necessary, given the raw configured
             # values. maxmem, memory, and shadow are all in KiB.
             # but memory_static_max etc are all stored in bytes now.
+            log.debug("_initDomain: setting memory_static_max to total_memory")                
+            self.info['memory_static_max'] = xc.physinfo()['total_memory'] * 1024
             memory = self.image.getRequiredAvailableMemory(
                 self.info['memory_dynamic_max'] / 1024)
             maxmem = self.image.getRequiredAvailableMemory(
