diff -up xen-3.4.0/tools/Makefile.dump xen-3.4.0/tools/Makefile
--- xen-3.4.0/tools/Makefile.dump	2009-05-18 13:05:38.000000000 +0200
+++ xen-3.4.0/tools/Makefile	2009-05-20 17:03:26.000000000 +0200
@@ -46,7 +46,7 @@ all: subdirs-all
 
 .PHONY: install
 install: subdirs-install
-	$(INSTALL_DIR) $(DESTDIR)/var/xen/dump
+	$(INSTALL_DIR) $(DESTDIR)/var/lib/xen/dump
 	$(INSTALL_DIR) $(DESTDIR)/var/log/xen
 	$(INSTALL_DIR) $(DESTDIR)/var/lib/xen
 	$(INSTALL_DIR) $(DESTDIR)/var/lock/subsys
--- xen-4.0.0/tools/python/xen/xend/XendDomainInfo.py.orig	2010-02-02 20:43:01.000000000 +0000
+++ xen-4.0.0/tools/python/xen/xend/XendDomainInfo.py	2010-02-02 21:36:57.000000000 +0000
@@ -2287,7 +2287,7 @@
             # To prohibit directory traversal
             based_name = os.path.basename(self.info['name_label'])
             
-            coredir = "/var/xen/dump/%s" % (based_name)
+            coredir = "/var/lib/xen/dump/%s" % (based_name)
             if not os.path.exists(coredir):
                 try:
                     mkdir.parents(coredir, stat.S_IRWXU)
@@ -2296,7 +2296,7 @@
 
             if not os.path.isdir(coredir):
                 # Use former directory to dump core
-                coredir = '/var/xen/dump'
+                coredir = '/var/lib/xen/dump'
 
             this_time = time.strftime("%Y-%m%d-%H%M.%S", time.localtime())
             corefile = "%s/%s-%s.%s.core" % (coredir, this_time,
