--- xen-4.1.1/tools/Makefile.orig2	2013-03-18 23:46:31.132000000 +0100
+++ xen-4.1.1/tools/Makefile	2013-03-18 23:47:14.936000000 +0100
@@ -2,50 +2,14 @@
 include $(XEN_ROOT)/tools/Rules.mk
 
 SUBDIRS-y :=
-SUBDIRS-y += check
 SUBDIRS-y += include
 SUBDIRS-y += libxc
-SUBDIRS-y += flask
 SUBDIRS-y += xenstore
-SUBDIRS-y += misc
-SUBDIRS-y += examples
 SUBDIRS-y += hotplug
-SUBDIRS-y += xentrace
-SUBDIRS-$(CONFIG_XCUTILS) += xcutils
-SUBDIRS-$(CONFIG_X86) += firmware
-SUBDIRS-$(ACM_SECURITY) += security
-SUBDIRS-y += console
-SUBDIRS-y += xenmon
-SUBDIRS-$(VTPM_TOOLS) += vtpm_manager
-SUBDIRS-$(VTPM_TOOLS) += vtpm
-SUBDIRS-y += xenstat
-SUBDIRS-$(CONFIG_Linux) += libaio
-SUBDIRS-$(CONFIG_Linux) += memshr 
-SUBDIRS-$(CONFIG_Linux) += blktap
-SUBDIRS-$(CONFIG_Linux) += blktap2
-SUBDIRS-$(CONFIG_NetBSD) += libaio
-SUBDIRS-$(CONFIG_NetBSD) += blktap2
-SUBDIRS-$(CONFIG_NetBSD) += xenbackendd
-SUBDIRS-y += libfsimage
-SUBDIRS-$(LIBXENAPI_BINDINGS) += libxen
-
-# do not recurse in to a dir we are about to delete
-ifneq "$(MAKECMDGOALS)" "distclean"
-SUBDIRS-$(CONFIG_IOEMU) += ioemu-dir
-endif
-
-SUBDIRS-y += xenpmd
-SUBDIRS-y += libxl
-SUBDIRS-y += remus
-SUBDIRS-$(CONFIG_X86) += xenpaging
-SUBDIRS-$(CONFIG_X86) += debugger/gdbsx
-SUBDIRS-$(CONFIG_X86) += debugger/kdd
 
 # These don't cross-compile
 ifeq ($(XEN_COMPILE_ARCH),$(XEN_TARGET_ARCH))
 SUBDIRS-$(PYTHON_TOOLS) += python
-SUBDIRS-$(PYTHON_TOOLS) += pygrub
-SUBDIRS-$(OCAML_TOOLS) += ocaml
 endif
 
 # For the sake of linking, set the sys-root
--- xen-4.1.1/tools/python/setup.py.orig	2013-03-21 00:08:57.672000000 +0100
+++ xen-4.1.1/tools/python/setup.py	2013-03-21 00:12:31.712000000 +0100
@@ -119,7 +119,7 @@
                                     [ XEN_ROOT + "/tools/libxl/libxenlight.so" ],
                sources            = [ "xen/lowlevel/xl/xl.c", "xen/lowlevel/xl/_pyxl_types.c" ])
 
-modules = [ xc, xs, ptsname, acm, flask, xl ]
+modules = [ xc, xs, ptsname, acm ]
 if plat == 'SunOS':
     modules.extend([ scf, process ])
 if plat == 'Linux':
@@ -131,23 +131,6 @@
       packages        = ['xen',
                          'xen.lowlevel',
                          'xen.util',
-                         'xen.util.xsm',
-                         'xen.util.xsm.dummy',
-                         'xen.util.xsm.flask',
-                         'xen.util.xsm.acm',
-                         'xen.xend',
-                         'xen.xend.server',
-                         'xen.xend.xenstore',
-                         'xen.xm',
-                         'xen.web',
-                         'xen.sv',
-                         'xen.xsview',
-                         'xen.remus',
-
-                         'xen.xend.tests',
-                         'xen.xend.server.tests',
-                         'xen.xend.xenstore.tests',
-                         'xen.xm.tests'
                          ],
       ext_package = "xen.lowlevel",
       ext_modules = modules
