diff -r 4683409fac3d xen/common/page_alloc.c
--- a/xen/common/page_alloc.c	Sun Sep 04 23:49:56 2011 +0200
+++ b/xen/common/page_alloc.c	Thu Sep 08 00:51:24 2011 +0200
@@ -567,7 +567,7 @@ static void free_heap_pages(
     avail[node][zone] += 1 << order;
     total_avail_pages += 1 << order;
 
-    if ( opt_tmem )
+    if ( 1 )
         midsize_alloc_zone_pages = max(
             midsize_alloc_zone_pages, total_avail_pages / MIDSIZE_ALLOC_FRAC);
 
