--- xen-4.2.1/tools/libxl/libxl_create.c.orig	2012-12-17 16:01:09.000000000 +0100
+++ xen-4.2.1/tools/libxl/libxl_create.c	2013-03-29 11:14:59.859901690 +0100
@@ -502,6 +502,9 @@ retry_transaction:
     libxl__xs_mkdir(gc, t,
                     libxl__sprintf(gc, "%s/data", dom_path),
                     rwperm, ARRAY_SIZE(rwperm));
+    libxl__xs_mkdir(gc, t,
+                    libxl__sprintf(gc, "%s/error", dom_path),
+                    rwperm, ARRAY_SIZE(rwperm));
     if (info->type == LIBXL_DOMAIN_TYPE_HVM)
         libxl__xs_mkdir(gc, t,
             libxl__sprintf(gc, "%s/hvmloader/generation-id-address", dom_path),
