--- xen-4.1.0/tools/python/xen/lowlevel/xl/xl.c.orig	2011-07-25 10:58:40.259999990 -0400
+++ xen-4.1.0/tools/python/xen/lowlevel/xl/xl.c	2011-07-25 11:02:48.323999735 -0400
@@ -418,10 +418,14 @@
         return NULL;
 
     domname = libxl_domid_to_name(&self->ctx, domid);
-    ret = PyString_FromString(domname);
-    free(domname);
-
-    return ret;
+    if (domname) {
+        ret = PyString_FromString(domname);
+        free(domname);
+        return ret;
+    } else {
+        Py_INCREF(Py_None);
+        return Py_None;
+    }
 }
 
 static PyObject *pyxl_domain_shutdown(XlObject *self, PyObject *args)
