--- xen-4.1.2/tools/python/xen/xend/server/SrvDaemon.py.orig	2012-04-15 17:17:50.167887550 +0100
+++ xen-4.1.2/tools/python/xen/xend/server/SrvDaemon.py	2012-04-15 17:31:13.648842655 +0100
@@ -325,7 +325,8 @@
     def set_user(self):
         # Set the UID.
         try:
-            os.setuid(pwd.getpwnam(XEND_USER)[2])
+            if XEND_USER != "root":
+                os.setuid(pwd.getpwnam(XEND_USER)[2])
             return 0
         except KeyError:
             print >>sys.stderr, "Error: no such user '%s'" % XEND_USER
--- xen-4.1.2/tools/misc/xend.orig	2012-08-07 00:54:02.372867624 +0100
+++ xen-4.1.2/tools/misc/xend	2012-08-07 00:58:11.469753483 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python -Es
 #  -*- mode: python; -*-
 #============================================================================
 # Copyright (C) 2004 Mike Wray <mike.wray@hp.com>
