diff -Nru xen/xen-4.1.2/tools/ioemu-qemu-xen/net.c xen-hvm/xen-4.1.2/tools/ioemu-qemu-xen/net.c
--- xen/xen-4.1.2/tools/ioemu-qemu-xen/net.c	2012-03-06 03:35:01.406652023 +0100
+++ xen-hvm/xen-4.1.2/tools/ioemu-qemu-xen/net.c	2012-03-06 03:33:25.534652121 +0100
@@ -124,6 +124,12 @@
 
 static struct TAPState *head_net_tap;
 
+#ifdef CONFIG_STUBDOM
+#include "qemu-xen.h"
+int net_lwip_init(VLANState *vlan, const char *model, const char *name, const char *ip, const char *mask, const char *gateway);
+void lwip_dhcp_config(const char *param, const char *value);
+#endif
+
 /***********************************************************/
 /* network device redirectors */
 
@@ -1724,6 +1730,48 @@
         ret = 0;
     } else
 #endif
+#ifdef CONFIG_STUBDOM
+    if (!strcmp(device, "lwip")) {
+        char lwip_gw[16], lwip_ip[16], lwip_mask[16];
+        char *p_gw = NULL, *p_ip = NULL, *p_mask = NULL, *p_dns = NULL, *p_client_ip = NULL;
+        if (get_param_value(lwip_mask, sizeof(lwip_mask), "netmask", p)) {
+            p_mask = lwip_mask;
+        } else
+            p_mask = xenstore_read("qubes_netmask");
+        if (p_mask)
+	    lwip_dhcp_config("netmask", p_mask);
+
+        if (get_param_value(lwip_ip, sizeof(lwip_ip), "server_ip", p)) {
+            p_ip = lwip_ip;
+        } else
+            p_ip = xenstore_read("qubes_secondary_dns");
+
+        if (get_param_value(lwip_gw, sizeof(lwip_gw), "gw", p)) {
+            p_gw = lwip_gw;
+        } else
+            p_gw = xenstore_read("qubes_gateway");
+        if (p_gw)
+	    lwip_dhcp_config("gw", p_gw);
+
+        if (get_param_value(buf, sizeof(buf), "dns", p)) {
+            p_dns = buf;
+        } else
+            p_dns = xenstore_read("qubes_gateway");
+        if (p_dns)
+	    lwip_dhcp_config("dns1", p_dns);
+
+        if (get_param_value(buf, sizeof(buf), "client_ip", p)) {
+            p_client_ip = buf;
+        } else
+            p_client_ip = xenstore_read("qubes_ip");
+        if (p_client_ip)
+	    lwip_dhcp_config("ip", p_client_ip);
+
+        vlan->nb_host_devs++;
+        ret = net_lwip_init(vlan, device, name, p_ip, p_mask, p_gw);
+
+    } else
+#endif
 #ifdef _WIN32
     if (!strcmp(device, "tap")) {
         char ifname[64];
@@ -1837,6 +1881,9 @@
 #ifdef CONFIG_SLIRP
                                        ,"user"
 #endif
+#ifdef CONFIG_STUBDOM
+									   ,"lwip"
+#endif
 #ifdef CONFIG_VDE
                                        ,"vde"
 #endif
diff -Nru xen/xen-4.1.2/tools/ioemu-qemu-xen/vl.c xen-hvm/xen-4.1.2/tools/ioemu-qemu-xen/vl.c
--- xen/xen-4.1.2/tools/ioemu-qemu-xen/vl.c	2012-03-06 03:34:27.546651991 +0100
+++ xen-hvm/xen-4.1.2/tools/ioemu-qemu-xen/vl.c	2012-03-06 03:33:46.590651989 +0100
@@ -4069,6 +4069,13 @@
            "                connect the user mode network stack to VLAN 'n' and send\n"
            "                hostname 'host' to DHCP clients\n"
 #endif
+#ifdef CONFIG_STUBDOM
+           "-net lwip[,vlan=n][,name=str][,client_ip=IP][,server_ip=IP][,netmask=MASK][,gw=GATEWAY][,dns=DNS]\n"
+           "                connect Lightweight IP network stack to VLAN 'n' with DHCP server\n"
+           "                server_ip is address for DHCP server,\n"
+		   "                the reset are given to client via DHCP\n"
+		   "                configuration can be also read from xenstore (qubes_* keys) - not used currently\n"
+#endif
 #ifdef _WIN32
            "-net tap[,vlan=n][,name=str],ifname=name\n"
            "                connect the host TAP network interface to VLAN 'n'\n"
diff -Nru xen/xen-4.1.2/tools/ioemu-qemu-xen/xen-hooks.mak xen-hvm/xen-4.1.2/tools/ioemu-qemu-xen/xen-hooks.mak
--- xen/xen-4.1.2/tools/ioemu-qemu-xen/xen-hooks.mak	2012-03-06 03:34:27.550651991 +0100
+++ xen-hvm/xen-4.1.2/tools/ioemu-qemu-xen/xen-hooks.mak	2012-03-06 02:09:53.818651722 +0100
@@ -46,7 +46,7 @@
 CONFIG_AUDIO=
 OBJS += xenfbfront.o
 
-OBJS+= qubes-gui.o txrx-vchan.o double-buffer.o qubes-gui-keymap.o
+OBJS+= qubes-gui.o txrx-vchan.o double-buffer.o qubes-gui-keymap.o net_lwip.o dhcpserver.o
 LIBS+= -lvchan -lxenvchan
 CPPFLAGS += -I$(XEN_ROOT)/tools/vchan -I$(XEN_ROOT)/tools/libvchan -I$(XEN_ROOT)/tools/qubes-gui/include
 VPATH+=:$(XEN_ROOT)/tools/qubes-gui/gui-common:$(XEN_ROOT)/tools/qubes-gui/gui-agent-qemu
