--- xen-4.2.0/tools/firmware/hvmloader/Makefile.orig	2012-05-27 21:57:04.481812859 +0100
+++ xen-4.2.0/tools/firmware/hvmloader/Makefile	2012-06-02 18:52:44.935034128 +0100
@@ -48,7 +48,7 @@
 else
 CIRRUSVGA_ROM := ../vgabios/VGABIOS-lgpl-latest.cirrus.bin
 endif
-ETHERBOOT_ROMS := $(addprefix ../etherboot/ipxe/src/bin/, $(addsuffix .rom, $(ETHERBOOT_NICS)))
+ETHERBOOT_ROMS := $(addprefix /usr/share/ipxe/, $(addsuffix .rom, $(ETHERBOOT_NICS)))
 endif
 
 ROMS := 
--- xen-4.4.0/Config.mk.orig	2014-06-24 00:06:06.704838751 +0200
+++ xen-4.4.0/Config.mk	2014-06-24 00:07:09.715775232 +0200
@@ -241,7 +241,7 @@
 # Fri Aug 2 14:12:09 2013 -0400
 # Fix bug in CBFS file walking with compressed files.
 
-ETHERBOOT_NICS ?= rtl8139 8086100e
+ETHERBOOT_NICS ?= 10ec8139 8086100e
 
 # Specify which qemu-dm to use. This may be `ioemu' to use the old
 # Mercurial in-tree version, or a local directory, or a git URL.
--- xen-4.4.0/tools/firmware/Makefile.orig	2014-06-24 00:05:20.338168036 +0200
+++ xen-4.4.0/tools/firmware/Makefile	2014-06-24 00:05:48.607904064 +0200
@@ -14,7 +14,7 @@
 endif
 SUBDIRS-$(CONFIG_ROMBIOS) += rombios
 SUBDIRS-$(CONFIG_ROMBIOS) += vgabios
-SUBDIRS-$(CONFIG_ROMBIOS) += etherboot
+#SUBDIRS-$(CONFIG_ROMBIOS) += etherboot
 SUBDIRS-y += hvmloader
 
 ovmf-dir:
