# Don't crash due to weird PCI cards (Bug 767742)

diff -r fb8dd4c67778 tools/python/xen/util/pci.py
--- a/tools/python/xen/util/pci.py	Tue Dec 13 14:16:20 2011 -0500
+++ b/tools/python/xen/util/pci.py	Wed Dec 14 15:46:56 2011 -0500
@@ -1268,7 +1268,12 @@ class PciDevice:
             pass
 
     def get_info_from_sysfs(self):
-        self.find_capability(0x11)
+        try:
+            self.find_capability(0x11)
+        except PciDeviceParseError, err:
+            log.error("Caught '%s'" % err)
+            return False
+
         sysfs_mnt = find_sysfs_mnt()
         if sysfs_mnt == None:
             return False
