--- xen-4.6.0/tools/hotplug/Linux/init.d/xendriverdomain.in.orig	2016-06-23 21:24:11.144000000 +0200
+++ xen-4.6.0/tools/hotplug/Linux/init.d/xendriverdomain.in	2016-06-23 21:25:19.064000000 +0200
@@ -51,6 +51,7 @@
 do_start () {
 	echo Starting xl devd...
 	${sbindir}/xl devd --pidfile=$XLDEVD_PIDFILE $XLDEVD_ARGS
+	( sleep 1; ${sbindir}/xl-devd-trigger; ) &
 }
 do_stop () {
         echo Stopping xl devd...
--- /dev/null	2016-04-24 06:35:41.993845619 +0200
+++ xen-4.6.0/tools/hotplug/Linux/xl-devd-trigger	2016-06-23 20:44:13.167000000 +0200
@@ -0,0 +1,33 @@
+#!/usr/bin/python
+
+import xen.lowlevel.xs
+
+def trigger(xs, trans):
+    backend_kind_list = xs.ls(trans, 'backend')
+    if not backend_kind_list:
+        return
+    for backend_kind in xs.ls(trans, 'backend'):
+        for domid in xs.ls(trans, '/'.join(['backend', backend_kind])):
+            for devid in xs.ls(trans,
+                    '/'.join(['backend', backend_kind, domid])):
+                state_path = '/'.join(['backend', backend_kind, domid, devid,
+                    'state'])
+                # it is critical to have those two operations in the same
+                # xenstore transaction, otherwise we may loose a race with xl
+                #  devd and rollback state back to '2'
+                if xs.read(trans, state_path) == '2':
+                    xs.write(trans, state_path, '2')
+
+def main():
+    xs = xen.lowlevel.xs.xs()
+    success = False
+    while not success:
+        trans = xs.transaction_start()
+        try:
+            trigger(xs, trans)
+            success = xs.transaction_end(trans)
+        except xen.lowlevel.xs.Error:
+            pass
+
+if __name__ == '__main__':
+    main()
--- xen-4.6.0/tools/hotplug/Linux/Makefile.orig	2016-06-23 21:31:16.698000000 +0200
+++ xen-4.6.0/tools/hotplug/Linux/Makefile	2016-06-23 21:34:46.483000000 +0200
@@ -57,6 +57,7 @@
 	$(INSTALL_DATA) $(XENCOMMONS_SYSCONFIG) $(DESTDIR)$(SYSCONFIG_DIR)/xencommons
 	$(INSTALL_PROG) $(XENDRIVERDOMAIN_INITD) $(DESTDIR)$(INITD_DIR)
 	$(INSTALL_PROG) init.d/xen-watchdog $(DESTDIR)$(INITD_DIR)
+	$(INSTALL_PROG) xl-devd-trigger $(DESTDIR)$(sbindir)
 
 .PHONY: install-scripts
 install-scripts:
