--- a/tools/libxl/libxl_dm.c
+++ b/tools/libxl/libxl_dm.c
@@ -1164,8 +1164,11 @@ static int libxl__build_device_model_arg
                          disks[i].pdev_path, disk, disks[i].readwrite ? "off" : "on", format, dev_number);
             } else {
                 if (!disks[i].readwrite) {
-                    LIBXL__LOG(ctx, LIBXL__LOG_ERROR, "qemu-xen doesn't support read-only disk drivers");
-                    return ERROR_INVAL;
+                    // Skip readonly disks, since QEMU can't emulate them. The
+                    // domain might still access them via PV drivers.
+                    LIBXL__LOG(ctx, LIBXL__LOG_WARNING, "cannot emulate"
+                               " read-only disk format for %s", disks[i].vdev);
+                    continue;
                 }
 
                 if (disks[i].format == LIBXL_DISK_FORMAT_EMPTY) {
