--- a/tools/libxl/libxl_dom_suspend.c
+++ b/tools/libxl/libxl_dom_suspend.c
@@ -75,6 +75,8 @@ int libxl__domain_suspend_device_model(l
     uint32_t const domid = dsps->domid;
     const char *const filename = dsps->dm_savefile;
 
+    uint32_t dm_domid;
+
     switch (libxl__device_model_version_running(gc, domid)) {
     case LIBXL_DEVICE_MODEL_VERSION_QEMU_XEN_TRADITIONAL: {
         LOG(DEBUG, "Saving device model state to %s", filename);
@@ -83,6 +85,11 @@ int libxl__domain_suspend_device_model(l
         break;
     }
     case LIBXL_DEVICE_MODEL_VERSION_QEMU_XEN:
+        dm_domid = libxl_get_stubdom_id(CTX, domid);
+        if (dm_domid != 0 && libxl__stubdomain_version_running(gc, dm_domid) ==
+                LIBXL_STUBDOMAIN_VERSION_LINUX)
+            return 0;
+
         if (libxl__qmp_stop(gc, domid))
             return ERROR_FAIL;
         /* Save DM state into filename */
