--- a/tools/libxl/libxl_nic.c
+++ b/tools/libxl/libxl_nic.c
@@ -243,8 +243,17 @@ static void libxl__device_nic_add(libxl_
 
         DEVICE_ADD(nic, nics, domid, &nic_saved, COMPARE_DEVID, &d_config);
 
-        rc = libxl__dm_check_start(gc, &d_config, domid);
-        if (rc) goto out;
+        /*
+         * Qubes workaround: Don't require a qemu running in dom0. If the
+         * domain has PV drivers NIC hotpluging still works.
+         */
+        uint32_t dm_domid = libxl_get_stubdom_id(CTX, domid);
+        if (dm_domid == 0 ||
+            libxl__stubdomain_version_running(gc, dm_domid) !=
+            LIBXL_STUBDOMAIN_VERSION_LINUX) {
+            rc = libxl__dm_check_start(gc, &d_config, domid);
+            if (rc) goto out;
+        }
     }
 
     for (;;) {
