--- a/stubdom/Makefile.orig	2012-02-21 15:42:20.364000956 +0100
+++ b/stubdom/Makefile	2012-02-21 15:43:40.916001086 +0100
@@ -350,6 +350,11 @@
 	  ln -sf $(XEN_ROOT)/tools/libxc/include/*.h include/ && \
 	  ln -sf $(XEN_ROOT)/tools/libxc/*.c . && \
 	  ln -sf $(XEN_ROOT)/tools/libxc/Makefile . )
+	mkdir -p vchan-$(XEN_TARGET_ARCH)
+	[ -h vchan-$(XEN_TARGET_ARCH) ] || ( cd vchan-$(XEN_TARGET_ARCH) && \
+	  ln -sf $(XEN_ROOT)/tools/vchan/*.c . && \
+	  ln -sf $(XEN_ROOT)/tools/vchan/*.h . && \
+	  ln -sf $(XEN_ROOT)/tools/vchan/Makefile.stubdom ./Makefile )
 	mkdir -p libvchan-$(XEN_TARGET_ARCH)
 	[ -h libvchan-$(XEN_TARGET_ARCH) ] || ( cd libvchan-$(XEN_TARGET_ARCH) && \
 	  ln -sf $(XEN_ROOT)/tools/libvchan/*.c . && \
@@ -267,6 +272,15 @@
  libxc-$(XEN_TARGET_ARCH)/libxenguest.a: libxc-$(XEN_TARGET_ARCH)/libxenctrl.a
 
 #######
+# vchan
+#######
+
+.PHONY: vchan
+vchan: vchan-$(XEN_TARGET_ARCH)/libvchan.a
+vchan-$(XEN_TARGET_ARCH)/libvchan.a:
+	CPPFLAGS="$(TARGET_CPPFLAGS) -I$(XEN_ROOT)/tools/libs/evtchn/include -I$(XEN_ROOT)/tools/libs/gnttab/include -I$(XEN_ROOT)/tools/libs/toollog/include" CFLAGS="$(TARGET_CFLAGS)" $(MAKE) -C vchan-$(XEN_TARGET_ARCH)
+
+#######
 # libvchan
 #######
 
@@ -321,8 +335,8 @@
 
 .PHONY: ioemu-stubdom
 ioemu-stubdom: APP_OBJS=$(CURDIR)/ioemu/i386-stubdom/qemu.a $(CURDIR)/ioemu/i386-stubdom/libqemu.a $(CURDIR)/ioemu/libqemu_common.a
-ioemu-stubdom: mini-os-$(XEN_TARGET_ARCH)-ioemu lwip-$(XEN_TARGET_ARCH) libxc ioemu
+ioemu-stubdom: mini-os-$(XEN_TARGET_ARCH)-ioemu lwip-$(XEN_TARGET_ARCH) libxc ioemu vchan libvchan
-	DEF_CPPFLAGS="$(TARGET_CPPFLAGS)" DEF_CFLAGS="$(TARGET_CFLAGS)" DEF_LDFLAGS="$(TARGET_LDFLAGS)" MINIOS_CONFIG="$(CURDIR)/ioemu-minios.cfg" $(MAKE) DESTDIR= -C $(MINI_OS) OBJ_DIR=$(CURDIR)/$< LWIPDIR=$(CURDIR)/lwip-$(XEN_TARGET_ARCH) APP_OBJS="$(APP_OBJS)"
+	DEF_CPPFLAGS="$(TARGET_CPPFLAGS)" DEF_CFLAGS="$(TARGET_CFLAGS)" DEF_LDFLAGS="$(TARGET_LDFLAGS)" MINIOS_CONFIG="$(CURDIR)/ioemu-minios.cfg" $(MAKE) DESTDIR= -C $(MINI_OS) OBJ_DIR=$(CURDIR)/$< LWIPDIR=$(CURDIR)/lwip-$(XEN_TARGET_ARCH) APP_OBJS="$(APP_OBJS)" vchan=y
 
 .PHONY: caml-stubdom
 caml-stubdom: mini-os-$(XEN_TARGET_ARCH)-caml lwip-$(XEN_TARGET_ARCH) libxc cross-ocaml caml
--- a/extras/mini-os/Makefile	2011-06-14 18:03:43.000000000 +0200
+++ b/extras/mini-os/Makefile	2012-02-18 01:05:39.288000355 +0100
@@ -83,6 +83,9 @@
 APP_LDLIBS += -lpci
 APP_LDLIBS += -lz
 APP_LDLIBS += -lm
+ifeq ($(vchan),y)
+APP_LDLIBS += -L$(XEN_ROOT)/stubdom/vchan-$(XEN_TARGET_ARCH) -lvchan -L$(XEN_ROOT)/stubdom/libvchan-$(XEN_TARGET_ARCH) -lxenvchan
+endif
 LDLIBS += -lc
 endif
 
